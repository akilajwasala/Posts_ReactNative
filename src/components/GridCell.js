import React from 'react';
import { TouchableOpacity, View, Dimensions, Image } from 'react-native';

const { width } = Dimensions.get('window');

export default  GridCell = ({ navigation, item, numColumns }) => {
    
    return (
        <TouchableOpacity onPress={() => {
            navigation.navigate('ImageView',{
                album: item,
                name: 'Image View'
            })
        }}>
            <View style={[styles.item, { height: width / numColumns}]}>
                <Image
                    style={[styles.imageStyle, { width: width / numColumns}]}
                    source={{ uri: item.thumbnailUrl }}
                />
            </View>
        </TouchableOpacity>
    )
}


const styles = {
    imageStyle: {
        height: null,
        flex: 1,
    },
    item: {
        alignItems: 'center',
        justifyContent: 'center',
        flex: 1,
        margin: 0
    },
    container: {
        flex: 1,
        marginVertical: 20,
    }
}