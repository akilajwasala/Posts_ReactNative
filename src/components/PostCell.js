import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';

export default PostCell = ({ navigation, item, name }) => {
    return (
        <TouchableOpacity onPress={() => {
                navigation.navigate('Details', {
                    post: item,
                    name: 'Image'
                })
            }}>
                <View style={styles.containerStyle}>
                    <Text style={styles.textStyleTitle}>{name}</Text>
                    <Text style={styles.textStyleNormal}>{item.title}</Text>
                </View>
        </TouchableOpacity>
    ) 
}

const styles = StyleSheet.create({
    containerStyle: {
        borderWidth: 1,
        borderRadius: 5,
        borderWidth: 1,
        borderColor: '#000000',
        borderBottomWidth: 1,
        shadowColor: '#000',
        shadowOffset: { width: 0, height: 2 },
        shadowOpacity: 0.1,
        shadowRadius: 2,
        elevation: 1,
        marginLeft: 5,
        marginRight: 5,
        marginTop: 10,
        justifyContent: 'flex-start',
        flex: 1,
        backgroundColor: '#6ED4C8'
    },
    textStyleNormal: {
        fontSize: 16,
        paddingTop: 10,
        paddingBottom: 10,
        marginLeft: 10,
        marginRight: 10
    },
    textStyleTitle: {
        fontSize: 18,
        paddingTop: 10,
        paddingBottom: 10,
        marginLeft: 10,
        marginRight: 10
    }
})