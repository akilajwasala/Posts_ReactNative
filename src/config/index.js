const env = 'dev'

const ENV_CONFIGURATION = {
    dev: {
        api_base: 'https://jsonplaceholder.typicode.com/',
    },
    prod: {
        api_base: 'https://jsonplaceholder.typicode.com/',
    }
};

const APPLICATIONS = {
    API_BASE: ENV_CONFIGURATION[env].api_base
}

export default APPLICATIONS