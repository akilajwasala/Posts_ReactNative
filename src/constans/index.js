import EndPoint from './endpoint';

export {
    EndPoint
}