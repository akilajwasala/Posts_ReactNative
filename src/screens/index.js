import Post from './Post/Posts';
import Details from './Post/Details';
import ImageView from './Post/ImageView';

export {
    Post,
    Details,
    ImageView
}