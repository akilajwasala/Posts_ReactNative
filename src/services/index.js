import { getPostAsync } from './post';
import { getUserAsync } from './user';
import { getPhotoAsync } from './photo';

export {
    getPostAsync,
    getUserAsync,
    getPhotoAsync
}