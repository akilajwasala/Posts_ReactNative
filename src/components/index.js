import PostCell from './PostCell';
import GridCell from './GridCell';

export {
    PostCell,
    GridCell
}