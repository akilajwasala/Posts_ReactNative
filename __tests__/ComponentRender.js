// __tests__/Posts.js
import React from 'react';
import { GridCell, PostCell } from '../src/components';

import mock from './mock';

import renderer from 'react-test-renderer';

//passing test case 
test('GridCell renders correctly', () => {
  const tree = renderer.create(
  <GridCell 
    navigation={mock.gridCell.navigation}
    item={mock.gridCell.item}
    numColumns={mock.gridCell.numColumns}
  />
).toJSON();
  expect(tree).toMatchSnapshot();
});


//failing test case
test('PostCell renders correctly', () => {
  const tree = renderer.create(
  <PostCell 
    navigation={mock.postCell.navigation}
    item={mock.postCell.item}
    name={mock.postCell.name}
  />
).toJSON();
  expect(tree).toMatchSnapshot();
});