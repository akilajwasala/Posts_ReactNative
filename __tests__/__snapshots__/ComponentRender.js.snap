// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridCell renders correctly 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "margin": 0,
        },
        Object {
          "height": 250,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "height": null,
          },
          Object {
            "width": 250,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`PostCell renders correctly 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#6ED4C8",
        "borderBottomWidth": 1,
        "borderColor": "#000000",
        "borderRadius": 5,
        "borderWidth": 1,
        "elevation": 1,
        "flex": 1,
        "justifyContent": "flex-start",
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 2,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "marginLeft": 10,
          "marginRight": 10,
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginLeft": 10,
          "marginRight": 10,
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    />
  </View>
</View>
`;
